### ISSUE TITLE
[MSan] False positive with `-fextend-variable-liveness=all`

### ISSUE THREAD

**Issue by dandrestor** [llvm/llvm-project#225425]
Hello! Consider the following test program being compiled with Clang 22.1.8. The variable `x` is never read, so it being uninitialized should not be a problem for MemorySanitizer.

`$ cat test.c`

```c
int main()
{
  unsigned char x;
  return 0;
}
```

```console
$ clang --version
clang version 22.1.8
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /nix/store/y97bdnihznxspkwhpcxzjpkivmrp99rc-clang-22.1.8/bin
```

When compiling with `-fextend-variable-liveness=all`, it produces spurious MSan failures:

```console
$ clang -fsanitize=memory -fextend-variable-liveness=all test.c -o test
$ ./test
==5484==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x5555556297b3  (/home/dan/test+0xd57b3)
    #1 0x7ffff7c2b284  (/nix/store/rkiwrqh8g6gs5rws4bf1s3mrdcs15gm9-glibc-2.42-84/lib/libc.so.6+0x2b284) (BuildId: efb85690f4b57f8e3536cf17e8835f0ec7631404)
    #2 0x7ffff7c2b337  (/nix/store/rkiwrqh8g6gs5rws4bf1s3mrdcs15gm9-glibc-2.42-84/lib/libc.so.6+0x2b337) (BuildId: efb85690f4b57f8e3536cf17e8835f0ec7631404)
    #3 0x5555555872f4  (/home/dan/test+0x332f4)

SUMMARY: MemorySanitizer: use-of-uninitialized-value (/home/dan/test+0xd57b3) 
Exiting
```

I would expect this to not produce any failures, as is the case when variable-liveness extension is disabled:

```console
$ clang -fsanitize=memory -fextend-variable-liveness=none test.c -o test
$ ./test
```

I was also able to reproduce this with Clang 23.1.1.
